tools/cellxgene_census_builder/src/cellxgene_census_builder/build_state.py

```diff
@@ -20,38 +20,70 @@
 
 CENSUS_BUILD_CONFIG = "config.yaml"
 CENSUS_BUILD_STATE = "state.yaml"
-CENSUS_CONFIG_DEFAULTS = {
-    # General config
-    "verbose": 1,
-    "log_dir": "logs",
-    "log_file": "build.log",
-    "reports_dir": "reports",
-    "consolidate": True,
-    "disable_dirty_git_check": True,
-    "dryrun": False,  # if True, will disable copy of data/logs/reports/release.json to S3 buckets. Will NOT disable local build, etc.
+
+
+@define
+class CensusBuildConfig:
+    verbose: int = field(converter=int, default=1)
+    log_dir: str = field(default="logs")
+    log_file: str = field(default="build.log")
+    reports_dir: str = field(default="reports")
+    consolidate = field(converter=bool, default=True)
+    disable_dirty_git_check = field(converter=bool, default=True)
+    dryrun: bool = field(
+        converter=bool, default=False
+    )  # if True, will disable copy of data/logs/reports/release.json to S3 buckets. Will NOT disable local build, etc.
     #
     # Paths and census version name determined by spec.
-    "cellxgene_census_S3_path": "s3://cellxgene-data-public/cell-census",
-    "logs_S3_path": "s3://cellxgene-data-public-logs/builder",
-    "build_tag": datetime.now().astimezone().date().isoformat(),
+    cellxgene_census_S3_path: str = field(default="s3://cellxgene-data-public/cell-census")
+    cellxgene_census_S3_remote_path: str = field(default=None)
```

Comment:
Looking forward, we may very well have multiple replicas.  May want to future-proof now if not expensive to build & test.

Reply:
I was thinking that the replicas would be managed by separate SFN steps, and each one will set a different value of `cellxgene_census_S3_remote_path`. This way we can easily run them in parallel.

Reply:
gotcha. I now understand!